Packages/ClientRuntime/Sources/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -68,38 +68,16 @@ public class CRTClientEngine: HttpClientEngine {
         return connectionPool
     }
     
-    public func executeWithClosure(request: SdkHttpRequest, completion: @escaping NetworkResult) {
-        execute(request: request).then { (result) in
-            completion(result)
-        }
-    }
-    
-    public func execute(request: SdkHttpRequest) -> Future<HttpResponse> {
+    public func execute(request: SdkHttpRequest) async throws -> HttpResponse {
         let connectionMgr = getOrCreateConnectionPool(endpoint: request.endpoint)
-        let httpResponseFuture: Future<HttpResponse> = connectionMgr.acquireConnection()
-            .chained { (connectionResult) -> Future<HttpResponse> in
-                self.logger.debug("Connection was acquired to: \(String(describing: request.endpoint.url?.absoluteString))")
-                let (requestOptions, future) = self.makeHttpRequestStreamOptions(request)
-                switch connectionResult {
-                case .failure(let error):
-                    future.fail(error)
-                case .success(let connection):
-                    let stream = connection.makeRequest(requestOptions: requestOptions)
-                    stream.activate()
-                    // map status code once call comes back
-                    future.then { (responseResult) in
-                        _ = responseResult.map { (response) -> HttpResponse in
-                            self.logger.debug("Future of response came back with success")
-                            let statusCode = Int(stream.getResponseStatusCode())
-                            response.statusCode = HttpStatusCode(rawValue: statusCode) ?? HttpStatusCode.notFound
-                            return response
-                        }
-                    }
-                }
-                
-                return future
-            }
-        return httpResponseFuture
+        let connection = try await connectionMgr.acquireConnection()
```

Comment:
In reading the implementation of the `acquireConnection()` function, it seems that the future will be called with either `fulfill` or `fail` -- therefore, it seems possible for this to fail, but it seems that we are always logging "Connection as acquired to: ...` is this the expected behavior?

Reply:
no that would be impossible. if acquireConnection failed it would throw and the function would throw from this line and never get to the next line.

Reply:
Oup, nevermind -- i was reviewing an older version of `acquireConnection()`. I didn't noticed it changed from `acquireConnection() -> Future<HttpClientConnection>` to `acquireConnection() async throws`